mediatek: add support for Cudy TR3000  256MB  v1 flash version

aiyunyuner opened the pull request:
This device is similar to the Cudy TR3000 v1 128MB version. The difference is that the flash memory is 128mb and the other is 256mb

Hardware:
 - SoC: MediaTek MT7981B
 - CPU: 2x 1.3 GHz Cortex-A53
 - Flash: 256 MiB SPI NAND
 - RAM: 512 MiB
 - WLAN: 2.4 GHz, 5 GHz (MediaTek MT7976CN, 802.11ax)
 - Ethernet: 1x 10/100/1000/2500 Mbps RTL8221B WAN, 1x10/100/1000 Mbps MT7981 LAN
 - USB 3.0 port
 - Buttons: 1 Reset button, 1 slider button
 - LEDs: 1x Red, 1x White
 - Power: 5 VDC, 3 A

Installation:
Cudy has distributed intermediate firmware to make installation easier
1. Go to [Cudy CN official website](https://www.cudy.com/zh-cn/pages/download-center/tr3000-1-0) and download the intermediate firmware
3. Upgrade the intermediate firmware on the page
4. Visit the intermediate firmware 192.168.1.1 webpage and use the sysupgrade image to update

other：
If you fail to flash the device, you can use TFTP to flash back to the original firmware.
1. Ask Cudy CN official customer service for the original firmware
2. With the router off, press the RESET button. While the router is turning on, the button should continue to be pressed for at least 5 seconds.
3. A u-boot shell will automatically open.
4. Connect to LAN and set your IP to 192.168.1.88/24. Configure a TFTP server and an recovery.bin firmware file.

csharper2005 commented on target/linux/mediatek/base-files/lib/preinit/05_set_preinit_iface:
```diff
@@ -3,6 +3,7 @@ set_preinit_iface() {
 	cudy,m3000-v1|\
 	cudy,tr3000-v1|\
 	cudy,tr3000-v1-ubootmod|\
+	cudy,tr3000-256mb-v1|\
```
Sorting.

csharper2005 reviewed (commented):
-

hauke commented:
Please squash the commits together.

I can not find the `R103` on https://www.cudy.com/de-de/pages/download-center/tr3000-1-0 . Is it not already there?

grinderz commented:
> I can not find the `R103` on https://www.cudy.com/de-de/pages/download-center/tr3000-1-0 . Is it not already there?

256MB offically available only in China, DTS and firmware download on China page  https://www.cudy.com/zh-cn/pages/download-center/tr3000-1-0

![Screenshot 2025-06-22 at 01 03 45](https://github.com/user-attachments/assets/2f364a6a-6c99-4cca-a44f-21e82524b5f3)
![Screenshot 2025-06-22 at 01 04 08](https://github.com/user-attachments/assets/6eb421ae-b059-4d9b-a398-ec3c194c0a45)

aiyunyuner committed:
mediatek: add support for Cudy TR3000  256MB  v1 flash version

This device is similar to the Cudy TR3000 v1 128MB version.
The difference is that the flash memory is 128mb and the other is 256mb

Hardware:
 - SoC: MediaTek MT7981B
 - CPU: 2x 1.3 GHz Cortex-A53
 - Flash: 256 MiB SPI NAND
 - RAM: 512 MiB
 - WLAN: 2.4 GHz, 5 GHz (MediaTek MT7976CN, 802.11ax)
 - Ethernet: 1x 10/100/1000/2500 Mbps RTL8221B WAN, 1x10/100/1000 Mbps MT7981 LAN
 - USB 3.0 port
 - Buttons: 1 Reset button, 1 slider button
 - LEDs: 1x Red, 1x White
 - Power: 5 VDC, 3 A

Installation:
Cudy has distributed intermediate firmware to make installation easier
1. Go to [Cudy CN official website](https://www.cudy.com/zh-cn/pages/download-center/tr3000-1-0) and download the intermediate firmware
2. Upgrade the intermediate firmware on the page
3. Visit the intermediate firmware 192.168.1.1 webpage and use the sysupgrade image to update

other:
If you fail to flash the device, you can use TFTP to flash back to the original firmware.
1. Ask Cudy CN official customer service for the original firmware
2. With the router off, press the RESET button. While the router is turning on, the button should continue to be pressed for at least 5 seconds.
3. A u-boot shell will automatically open.
4. Connect to LAN and set your IP to 192.168.1.88/24. Configure a TFTP server and an recovery.bin firmware file.

Signed-off-by: cheng wang <typedelta@outlook.com>
Link: https://github.com/openwrt/openwrt/pull/19167
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

aiyunyuner commented:
> Please squash the commits together.
OK

hauke commented:
> > I can not find the `R103` on https://www.cudy.com/de-de/pages/download-center/tr3000-1-0 . Is it not already there?
> 
> 256MB offically available only in China, DTS and firmware download on China page https://www.cudy.com/zh-cn/pages/download-center/tr3000-1-0
> 
> ![Screenshot 2025-06-22 at 01 03 45](https://private-user-images.githubusercontent.com/10777697/457609148-2f364a6a-6c99-4cca-a44f-21e82524b5f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA2MTIxMzQsIm5iZiI6MTc1MDYxMTgzNCwicGF0aCI6Ii8xMDc3NzY5Ny80NTc2MDkxNDgtMmYzNjRhNmEtNmM5OS00Y2NhLWE0NGYtMjFlODI1MjRiNWYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjIyVDE3MDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlZDBlNzhlNGQxN2VlMjkzMWY5NGQwNzEyMjUxNGI2NDg0ZjNhNmUzOTVlMzYzY2I4YzIzMjZkOTBiYzg5ODMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gqNHxhScIVBN9LRa5mc59wr-HXQLcQqwdbspJVE1Bzs) ![Screenshot 2025-06-22 at 01 04 08](https://private-user-images.githubusercontent.com/10777697/457609150-6eb421ae-b059-4d9b-a398-ec3c194c0a45.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTA2MTIxMzQsIm5iZiI6MTc1MDYxMTgzNCwicGF0aCI6Ii8xMDc3NzY5Ny80NTc2MDkxNTAtNmViNDIxYWUtYjA1OS00ZDliLWEzOTgtZWMzYzE5NGMwYTQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA2MjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNjIyVDE3MDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiNWNmNjgwNDM5OTE4OTFiNDAyZmZlZTkzODJmNmFjZTdjNTA2ZWEzMTFhZWFiZGQxNWYzMDczNjU0YjZhOTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZyhcDMy_DIv3Go3QPeDpNIhgg3QXtX8LQZ6g2FCeihg)

Thank you for the screen shots, it forwards me alleyways to the German website.

grinderz commented:
> it forwards me alleyways

yup they ignore in url language selection, need to select manually in top right

![Screenshot 2025-06-22 at 20 29 03](https://github.com/user-attachments/assets/1840df70-94a1-47dc-baa4-1a87d41f3c60)

fildunsky commented:
Hey, thanks for your PR. I've just [built vanilla 24.10.2](https://github.com/fildunsky/Fudy-MT3000/raw/refs/heads/main/OpenWRT%2024.10.2/openwrt-24.10.2-mediatek-filogic-cudy_tr3000-256mb-v1-squashfs-sysupgrade.bin) out of it.

ZsuZmin commented:
我编译了您的tr3000 256mb固件，lici后台配置有wifi选项，但是搜索不到wifi信号，需要配置什么吗？

aiyunyuner commented:
> 我编译了您的tr3000 256mb固件，lici后台配置有wifi选项，但是搜索不到wifi信号，需要配置什么吗？

2.4g的wifi可以用吗？

aiyunyuner commented:
> Hey, thanks for your PR. I've just [built vanilla 24.10.2](https://github.com/fildunsky/Fudy-MT3000/raw/refs/heads/main/OpenWRT%2024.10.2/openwrt-24.10.2-mediatek-filogic-cudy_tr3000-256mb-v1-squashfs-sysupgrade.bin) out of it.

thanks，i will try it later

aiyunyuner commented:
> By the way, will you be able to add same PR as a backport to 24.10 branch so it will become available in next builds like 24.10.3? Otherwise it will not happen automatically.

Do i need to submit another PR on 24.10?

fildunsky commented:
> > By the way, will you be able to add same PR as a backport to 24.10 branch so it will become available in next builds like 24.10.3? Otherwise it will not happen automatically.
> 
> Do i need to submit another PR on 24.10?

Yes, same code, but tag it for 24.10 as a backport

foi commented:
> Hey, thanks for your PR. I've just [built vanilla 24.10.2](https://github.com/fildunsky/Fudy-MT3000/raw/refs/heads/main/OpenWRT%2024.10.2/openwrt-24.10.2-mediatek-filogic-cudy_tr3000-256mb-v1-squashfs-sysupgrade.bin) out of it.

I've Successfully upload your firmware and it works :)!
![2025-07-03-113608_hyprshot](https://github.com/user-attachments/assets/d1ca12ff-ccce-4c6f-902c-25f587432867)

aiyunyuner commented:
> > > By the way, will you be able to add same PR as a backport to 24.10 branch so it will become available in next builds like 24.10.3? Otherwise it will not happen automatically.
> > 
> > 
> > Do i need to submit another PR on 24.10?
> 
> Yes, same code, but tag it for 24.10 as a backport

thanks，I add same PR to 24.10 branch

777vip commented:
> @777vip我帮你建好了，兄弟https://github.com/fildunsky/Fudy-MT3000/blob/main/OpenWRT%2024.10.2/openwrt-24.10.2-mediatek-filogic-cudy_tr3000-256mb-v1-squashfs-sysupgrade.bin是纯版本OpenWrt 24.10.2，可以从官方仓库安装所有官方内核模块。你需要先从Cudy中文官网刷入过渡版OpenWrt固件架构，然后才能通过网页界面刷入我的文件。WiFi默认关闭，所以你需要用网线来设置你的路由器……

非常感谢这位兄弟回答我的问题。

777vip commented:
> > > > 顺便问一下，您能否将相同的 PR 添加为 24.10 分支的反向移植，以便它在下一个版本（例如 24.10.3）中可用？否则它不会自动发生。
> > > 
> > > 
> > > 我需要在 10 月 24 日提交另一个 PR 吗？
> > 
> > 
> > 是的，相同的代码，但将其标记为24.10作为反向移植
> 
> 谢谢，我把同样的 PR 添加到 24.10 分支

您有安装ImmortalWrt固件的教程吗？
我自己尝试过先刷cudy官方给的中间过渡包，在刷ImmortalWrt的immortalwrt-mediatek-filogic-cudy_tr3000-256mb-v1-squashfs-sysupgrade.bin
但是无法开机。
您知道这是怎么回事吗？
是方法不正确吗？

777vip commented:
> @777vip因为官方的 24.10.2 版本发布之前，256MB 的代码还没有写完。一旦像 24.10.2 这样的稳定​​版本发布，它就基本上沿用了发布时的代码。所有新代码都会先转到主分支（称为快照）。之后，人们通常会将代码反向移植到 24.10 分支，如果在那里被接受，我们就必须等待下一个稳定版本，比如 24.10.3。OpenWRT 以前就是这样的。

谢谢您的解答，祝您发财。
